lib/voxpupuli/release/rake_tasks.rb

```diff
@@ -52,3 +52,90 @@
     fail "Unable to find a CHANGELOG.md entry for the #{v} release."
   end
 end
+
+desc "Prepare a release"
+task "release:prepare" do
+  Rake::Task["release:porcelain:changelog"].invoke
+  Rake::Task["strings:generate:reference"].invoke if File.exist?('REFERENCE.md')
+end
+
+begin
+  require 'github_changelog_generator/task'
+  require 'puppet_blacksmith'
+
+  class GCGConfig
+    def self.user=(user)
+      @user = user
+    end
+
+    def self.user
+      @user || metadata['name'].split(%r{[-/]}).first
+    end
+
+    def self.metadata
+      @metadata ||= Blacksmith::Modulefile.new.metadata
+    end
+  end
+
+  task "release:porcelain:changelog" do
+    # This is taken from lib/github_changelog_generator/task
+    # The generator cannot be used because we want to lazyly evaluate
+    # GCGConfig.user which might be overrider in the module Rakefile.
+    options = GitHubChangelogGenerator::Parser.default_options
+    options[:user] = GCGConfig.user
+    options[:project] = GCGConfig.metadata['name']
+    options[:future_release] = "v#{GCGConfig.metadata['version']}" if GCGConfig.metadata['version'] =~ /^\d+\.\d+.\d+$/
+    options[:header] = <<~HEADER.chomp
```

Comment:
Is `.chomp` needed? I'd expect that with `<<~` it's redundant.

Reply:
Still seems required:
```sh-session
romain@desktop-fln40kq ~/Projects/voxpupuli/voxpupuli-release % cat /tmp/foo.rb
puts <<~DATA.inspect
moin
DATA
romain@desktop-fln40kq ~/Projects/voxpupuli/voxpupuli-release % ruby /tmp/foo.rb
"moin\n"
```